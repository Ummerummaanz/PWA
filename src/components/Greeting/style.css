.feature-header-toolbar__greeting__message {
  font-size: 1em;
  color: var(--feature-header-toolbar-greeting-message);
}
