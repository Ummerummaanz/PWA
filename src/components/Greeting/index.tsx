import React from 'react';
import { Translate } from 'oasis-os-common';
import { useAppState } from 'oasis-os-react';
import './style.css';

interface Props {
  message: string;
}

const Greeting: React.FC<Props> = ({ message }: Props) => {
  const [, , app] = useAppState();
  const entryId = app?.meta?.contentId as string;
  return (
    <div className="feature-header-toolbar__greeting__message" data-testid="greeting_message">
      <Translate id={`${entryId}.greetingMessage`} />&nbsp;{message}
    </div>
  );
};

export default Greeting;
