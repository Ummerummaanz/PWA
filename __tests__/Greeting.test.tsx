import * as React from 'react';
import { render } from '@testing-library/react';
import Greeting from '../src/components/Greeting';

jest.mock('oasis-os-common', () => ({
  ...jest.requireActual('oasis-os-common'),
  Translate: jest.fn(({ id }) => {
    return "hello";
  }),
}));

jest.mock('oasis-os-react', () => ({
  ...jest.requireActual('oasis-os-react'),
  dispatch: jest.fn(),
  useAppState: jest.fn(() => {
    return [
      undefined,
      undefined,
      {
        meta: { contentId: 'foo' },
      },
    ];
  }),
}));

describe('Feature', () => {
  it('should have greeting message', () => {
    const { container, getByTestId } = render(<Greeting message="test" />);
    getByTestId('greeting_message');
    expect(container).toMatchSnapshot();
  });
});
