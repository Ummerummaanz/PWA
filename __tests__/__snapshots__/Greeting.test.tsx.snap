// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature should have greeting message 1`] = `
<div>
  <div
    class="feature-header-toolbar__greeting__message"
    data-testid="greeting_message"
  >
    test
  </div>
</div>
`;
